File: eventapp/ClientApp/app/models/event.ts
import { EventType } from './enums/EventTypes.enum';

export default class Event {
    private type: EventType = EventType.Event;
    id: number;
    title: string;
    description: string;
    createdByUsername: string;
    startDate: Date;
    endDate: Date;
    reminder: boolean;
    publicEvent: boolean;
}

File: eventapp.Domain/Models/Event.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace eventapp.Domain.Models
{
    public class Event
    {
        [Key]
        public long? Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public string CreatedBy { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool Reminder { get; set; }
        public bool publicEvent { get; set; }
        public bool ReminderSent { get; set; }

    }
}
